### DISCUSSION TITLE
UsageStats: Improve plugin reporting

### DISCUSSION THREAD

**Discussion by RichiH** [grafana/grafana#38321] [Plugins]
I think we should enhance our plugin reporting to get a better understanding of what our users need and want. We need to do this in a way which still preserves privacy for private plugins and the like.

The plan is to
* keep reporting everything on current allowlist
* introduce a new label-based system to set plugins to public as part of their own config
* set all the plugins we provide and/or host to public
* add documentation, suggesting people set their plugins to public so we can spread love for the ones gaining traction
report everything set to public on top of the allowlist
* add the same to howtos, templates, etc.

The above ensures that we still default to privacy, but extract better signal from public information to get more insight into what our users do. Personally, I think this strikes a better balance as it will allow us to understand our users better.

CC @bergquist @daniellee @torkelo

**Comment by torkelo** [comment]
This should be pretty straight forward, we need to add a boolean to plugin.json , something like "usageStats: true/false" , and then check that here: https://github.com/grafana/grafana/blob/master/pkg/infra/usagestats/usage_stats.go#L94 


throughts @marefr @wbrowne ?

**Comment by wbrowne** [comment]
Yep, I think this is a good idea! It would be great to come up with a list of different signals in order to help the goal of 
>understanding of what our users need and want

This would be a good start

**Comment by RichiH** [comment]
I would be less concerned about getting it perfect than just getting started; we can always iterate on top. :)

**Comment by RichiH** [comment]
CC @marcusolsson as a central point for updating documentation, howtos, etc.

**Comment by marefr** [comment]
Sounds good to me

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
